src/components/Palette/Palette.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+
+import {useSelect} from '../../hooks';
+import {useForkRef} from '../../hooks/useForkRef/useForkRef';
+import type {ButtonProps} from '../Button';
+import {Button} from '../Button';
+import type {ControlGroupProps, DOMProps, QAProps} from '../types';
+import {block} from '../utils/cn';
+
+import {usePaletteGrid} from './hooks';
+import {getPaletteRows} from './utils';
+
+import './Palette.scss';
+
+const b = block('palette');
+
+export type PaletteOption = Pick<ButtonProps, 'disabled' | 'title'> & {
+    /**
+     * Option value, which you can use in state or send to back-end and so on.
+     */
+    value: string;
+    /**
+     * Content inside the option (emoji/image/GIF/symbol etc).
+     *
+     * Uses `value` as default, if `value` is a number, then it is treated as a unicode symbol (emoji for example).
+     *
+     * @default props.value
+     */
+    content?: React.ReactNode;
+};
+
+export interface PaletteProps
+    extends Pick<ControlGroupProps, 'aria-label' | 'aria-labelledby'>,
+        Pick<ButtonProps, 'disabled' | 'size'>,
+        DOMProps,
+        QAProps {
+    /**
+     * Allows selecting multiple options.
+     *
+     * @default true
+     */
+    multiple?: boolean;
+    /**
+     * Current value (which options are selected).
+     */
+    value?: string[];
+    /**
+     * The control's default value. Use when the component is not controlled.
+     */
+    defaultValue?: string[];
+    /**
+     * List of Palette options (the grid).
+     */
+    options?: PaletteOption[];
+    /**
+     * How many options are there per row.
+     *
+     * @default 6
+     */
+    columns?: number;
+    /**
+     * HTML class attribute for a grid row.
+     */
+    rowClassName?: string;
+    /**
+     * HTML class attribute for a grid option.
+     */
+    optionClassName?: string;
+    /**
+     * HTML class attribute for the `<Button.Icon />` inside a grid option.
+     */
+    iconClassName?: string;
```

Comment:
`icon` name is misleading, icon is a way we center content in the Button. And why we need this class in the first place?

Reply:
I don't think we really need it then
Removed it